ekwipunek.py

```diff
@@ -0,0 +1,823 @@
+"""
+Konwencja -> Koncept jest taki, że w kodzie polskich znaków nie ma oprócz stringów.
+Wszystko w pierdolnikach jest oficjalne
+"""
+# komentarze zaczynajace sie od hashtaga sa skierowane bezposrednio do Ciebie
+from abc import ABC
+
+# no tu pewnie troche przesadzilem :D
+# pylint: disable=missing-function-docstring, too-many-arguments, unnecessary-pass
+# pylint: disable=too-few-public-methods, protected-access, too-many-branches
+# pylint: disable=too-many-nested-blocks, too-many-instance-attributes
+# pylint: disable=useless-parent-delegation
+class Przedmiot(ABC):
+    """
+    Base klasa, abstrakcyjna.
+
+    :param nazwa: nazwa przedmiotu
+    :param wartosc: nazwa przedmiotu wyrażana w sztukach złota
+    :param efekt: efekt to zamiennik pola uzycie, jak rowniez bonusy ->
+        jego obecnosc informuje iz przedmiot mozna uzyc, a sam efekt informuje
+        o tym co przedmiot robi
+    """
+
+    def __init__(self, nazwa: str, wartosc: int, efekt: dict[str, int] | None):
+        self._nazwa = nazwa
+        self._wartosc = wartosc
+        self._efekt = efekt
+    @property
+    def nazwa(self):
+        return self._nazwa
+
+    @property
+    def wartosc(self):
+        return self._wartosc
+
+    @property
+    def efekt(self):
+        return self._efekt
+
+
+class Bron(Przedmiot):
+    """
+    Klasa broni.
+
+    :param nazwa: nazwa przedmiotu
+    :param wartosc: wartość przedmiotu wyrażana w sztukach złota
+    :param wymagania: wymagane atrybuty do użycia danego przedmiotu
+    :param obrazenia: wartość zadawanych przez przedmiot obrażeń
+    :param efekt: efekt to zamiennik pola uzycie, jak rowniez bonusy ->
+        jego obecnosc informuje iz przedmiot mozna uzyc, a sam efekt informuje
+        o tym co przedmiot robi
+    """
+
+    def __init__(
+        self,
+        nazwa: str,
+        wartosc: int,
+        wymagania: dict[str, int],
+        obrazenia: float,
+        efekt: dict[str, int] | None = None,
+    ):
+        if efekt is None:
```

Comment:
tutaj możesz dodać else który mówi że jesli efekt jest dict to zupdatuj ten pole "obrazenia"